pennylane/devices/qubit/apply_operation.py

```diff
@@ -231,6 +237,8 @@ def apply_conditional(
     Returns:
         ndarray: output state
     """
+    mid_measurements = execution_kwargs.get("mid_measurements", None)
+
     if op.meas_val.concretize(mid_measurements):
         return apply_operation(
```

Comment:
We always pass all the `execution_kwargs`, but the dispatcher for `Conditional` only relies on `mid_measurements`. The RNG stuff is not needed since there is no non-deterministic behaviour expected when applying `Conditional` operations. Such behaviour is only expected when applying `MidMeasureMP`.